Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.SwitchExpression.cs

```diff
@@ -49,18 +50,22 @@ protected internal override Expression Accept (ExpressionVisitor visitor)
       return default(Expression);
     }
 
+    [Pure]
     public SwitchExpression Update (Expression switchValue, IEnumerable<SwitchCase> cases, Expression defaultBody)
     {
+      Contract.Requires(switchValue != null);
```

Comment:
For this and all similar cases:

I added `Contract.Requires(cases != null);`. For the methods that accept a params array, I also added `Contract.Requires(cases.Length >= 1);`. I did not add `Contract.Requires(cases.Any());` or `Contract.Requires(Contract.ForAll(cases, case => case != null);` because those cannot be proven by the static checker and it will be constant hassle for application developers. Perhaps we can revisit it when the static checker gets smarter.